### DISCUSSION TITLE
Disabling "click for fullscreen" behaviour for images and footnotes

### DISCUSSION THREAD

**Discussion by tilman-schieber** [LiaScript/LiaScript#195] [Q&A] [answered]
I have the requirement to use some pictures just for decoration on a course site so I do not want them to be clickable
Here ist my use case: [liascript editor](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA6XRwWuDMBgF8Lvg/5DWy3aIn5Vu3ew22Bg7FXbqaQyJ5kODMQlJrPW/n+IY3VF6fIcHP96LoogchOrOhElH3tBbgYVzg/PYhkEYPK0oJcT5QeLzumW2EipLsd0TLpyRbMiEkkIhLaQumz3pBfd1RjZ3iTmvCaUvYbD6Ohad8t33Te29cRlAZ6RmPO5FI1rkgsXaVjAlMyUoddtq5WAHu3uYu1TqStM0SdPYnarbK2CbS9g7FoKpxbAtbBnMXfppUB1G3JWuNL10vdqyltMpi2kI+AB/9Xk255nizHJ6QuuEVrFRV03479sP5Nqyxc4EkkeYu/mEzEdi3mvLR0bzu+UPHK7Q8ZsCAAA=)

I would love to use the gallery for this but because all the logos I am showing here have a transparent background, it looks really bad. Also I do not want to have the click-for-fullscreen behaviour. 
In general I would love an easy way to disable this behaviour for footnotes as well. A fullscreen footnote looks really weird if it's just a single sentence.

**Comment by andre-dietrich** [comment]
## Disabling clickable images

The image part is quite simple, try something like this:

```` markdown
### Linux als Betriebssystem

<!--  style="margin:2em; display:inline-block; width: 150px" -->
![Ubuntu](https://upload.wikimedia.org/wikipedia/commons/7/76/Ubuntu-logo-2022.svg)<!-- data="true" -->
````

The click and enlarge behavior for images is triggered, if the user has NOT defined some custom stiles/CSS. You can modify the width for example, change the background or define a random parameter like data-something ... 

## Gallery with background

But If you want to have a gallery and change the background, you can try this:

``` markdown
### Linux als Betriebssystem

![Ubuntu](https://upload.wikimedia.org/wikipedia/commons/7/76/Ubuntu-logo-2022.svg "Ubuntu")<!-- style="background-color: lightgray" -->
![Debian](https://upload.wikimedia.org/wikipedia/commons/4/4a/Debian-OpenLogo.svg "debian")<!-- style="background-color: lightgray" -->
![Archlinux](https://upload.wikimedia.org/wikipedia/commons/e/e8/Archlinux-logo-standard-version.png "Arch-Linus")<!-- style="background-color: lightgray" -->
![Fedora](https://upload.wikimedia.org/wikipedia/commons/0/09/Fedora_logo_and_wordmark.svg "Fedora")<!-- style="background-color: lightgray" -->
```

In this case, styling will be applied. But I don't know why the Ubuntu-image is not displayed at all ... The others work as expected ...

## Footnotes

Do you mean, that you only want to disable that clicking onto a footnote will show the associated text?

**Reply by tilman-schieber** [reply]
Regarding footnotes: in my mind the default behaviour should be jumping to a footnote (like Wikipedia). I am sure the modal dialogue style footnotes are great if you have footnotes with pictures or video, for simple annotations it feels clunky.

**Reply by andre-dietrich** [reply, marked as answer]
Footnote is a button, where the `onclick` function is defined in `window.LIA.showFootnote`. You can overwrite this function and implement your own behavior like this:

``` markdown
<!--

@onload
const showFootnote = window.LIA.showFootnote

window.LIA.showFootnote = (ref) => {
  if (window.LIA.settings.mode === "Textbook") {
    document.getElementById("footnote-" + ref).scrollIntoView()
  } else {
    showFootnote(ref)
  }
}
@end

-->

# Footnotes

This is a [^ref] with only one URL.

Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et
dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet
clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet,
consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea
takimata sanctus est Lorem ipsum dolor sit amet.

Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et
dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet
clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet,
consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea
takimata sanctus est Lorem ipsum dolor sit amet.

...

[^ref]: https://LiaScript.github.io
```

In Textbook mode this will scroll to the associated element, otherwise, if the footnotes are not visible, it will display them ...

[Demo: LiveEditor](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA+2XQWsbMRCF7/oVr+7FpvbmHuqQFFow+NSkvZQW5NV4rXqlcaWRXVPy3zvrOOBLeuohB8GyLDtvZp4e30Xv38xmxtxy7Nk603LMgrzhwydmiSyEOQ4+Oj40y8Vdc1kx5oWCtowTrSeY3+CPAfwa40spifjY5SawU+18jtED/ZYV83Y0OTUAjtsSKErTkXzsafj8cFy48Wh93jEb4R2GJU1uE/f9Igp/9XQYT7T/EdRnOo+6dHayNQjMo7ml6IyZzW6MeYtnQTbmYeMz9LH49kPl3/X4soHmc9QX4cvnZWOMWXKiAL/LJajZnhOyF9hAMsUQIglJ0Z/WqabV46olL2mKTA7O24DIsYQjyCfNAUJhp0N83HtXoqAIervSJSB5WkAItosWtve/ylEHULJyMW/PfdlJsQ3uBHtKDOI8dNu2LXnQC36WLAxX+DzyVCerQa5KaHCvptGqT4utzQ5dWVHqEsWputVNFmK3PlitZxtbKdqvuLycRfOPWs3pIqcKVAWqAlWBer05VaAqUBWoCtTrzakCVYH6z0CZpyvgNTYiu3x9dbX09r5NfqcXU70VllXj2fwF/htpGEIPAAA=)
